conf/environments.js

```diff
@@ -102,6 +102,11 @@ module.exports = {
             ecmaVersion: 6
         }
     },
+    es7: {
```

Comment:
If there won't be any new environments in the future along this naming convention, then it makes sense to stop at "6" - but there's already 3, 5, and 6 - the cowpath is pretty paved imo.

What happens when es5 is false and es6 is true? Isn't it the same thing, where es6 subsumes es5, so setting es5 to false has no effect? (like setting ecmaVersion, rather than like setting env flags)

That said, if when adding an "8" you'd also add a "7", certainly you'd avoid the confusion I described.

I'm most interested in the `space-infix-ops` and `espree` changes - if the env stuff is contentious I'd prefer to split the env stuff into a separate PR so I can start using `**` in a linted codebase :-)